tests/testthat/test-modspec.R

```diff
@@ -163,9 +149,192 @@ test_that("at options are parsed", {
   expect_warning(ats('@foo "a b c"'))  
 })
 
-test_that("specMATRIX", {
+test_that("HANDLEMATRIX", {
   code <- "$OMEGA 1,2,3"
   mod <- mcode("test-spec-matrix", code, compile = FALSE)
   mat <- unname(as.matrix(omat(mod)))
   expect_true(all.equal(mat, dmat(1,2,3)))
 })
+
+test_that("programmatic initialization", {
+  code <- '
+$ENV
+mat1 <- matrix(0,1,1)
+mat2 <- matrix(0,2,2)
+mat3 <- matrix(0,3,3)
+rownames(mat3) <- letters[1:3]
+mat4 <- matrix(0,4,4)
+par <- list(z = 777)
+pcmt <- c("t", "u", "v")
+
+$OMEGA
+1
+
+$OMEGA @object mat2 @name omega2
+
+$OMEGA @as_object 
+m <- matrix(0,3,3)
+rownames(m) <- LETTERS[1:3]
+m
+
+$SIGMA
+11
+
+$SIGMA @object mat4
+$SIGMA @as_object
+matrix(0,5,5)
+
+$PARAM @as_object
+list(a = 1, b = 2)
+
+$THETA @as_object @name theta
+rep(0,2)
+
+$PARAM @object par
+
+$CMT @as_object
+c("gg", "hh", "iii")
+
+$CMT @object pcmt
+'
+  
+  mod <- mcode("foo", code, compile = FALSE)
+  x <- labels(mod)
+  expect_equal(x$param, c("a", "b", "theta1", "theta2", "z"))
+  expect_equal(mod$z, 777)
+  expect_equal(x$omega_labels[[3]], c("A", "B", "C"))
+  expect_equal(x$omega[2], "omega2")
+  d <- nrow(omat(mod))
+  expect_equal(unname(nrow(omat(mod))), c(1,2,3))
+  expect_equal(unname(nrow(smat(mod))), c(1,4,5))
+  expect_equal(x$init, c("gg", "hh", "iii", "t", "u", "v"))
+})
+
+test_that("parse content using low-level handlers - PARAM", {
+  env <- mrgsolve:::parse_env(vector(mode = "list", length = 20), project = '.')
+  sup <- suppressMessages  
+  
+  input <- "c(1,2,3)"
+  expect_error(
+    sup(mrgsolve:::PARAM(x = input, as_object = TRUE)), 
+    "code returned the incorrect class: numeric"
+  )
+  input <- "list(a = 1, b = 2)"
+  ans <- mrgsolve:::PARAM(x = input, as_object = TRUE, env = env, pos = 3)
+  expect_is(env$param[[3]], "list")
+  expect_named(env$param[[3]])
+  
+  expect_null(env$param[[8]])  
+  env$ENV$parameters <- list(mm = 1, nn = 2)
+  ans <- mrgsolve:::PARAM(x = input, object = "parameters", env = env, pos = 8)
```

Comment:
In some of these tests where you pass an object name instead of setting `as_object = TRUE`, you set `x = ""`. In other tests (like this), `x` is passed an existing object. It looks like `object` is taking precedence over `x` for these. That makes sense, but I'm just wondering what would happen if I tried using `@object` and `@as_object` in the same block. Is there a check for that before it even gets to calling these functions?

Reply:
Thanks, Tim.  We'll create an error if both are supplied.